src/main/java/roomescape/domain/reservation/service/ThemeService.java

```diff
@@ -0,0 +1,67 @@
+package roomescape.domain.reservation.service;
+
+import java.time.Clock;
+import java.time.LocalDate;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+import roomescape.common.exception.AlreadyInUseException;
+import roomescape.domain.reservation.dto.ThemeRequest;
+import roomescape.domain.reservation.dto.ThemeResponse;
+import roomescape.domain.reservation.entity.Theme;
+import roomescape.domain.reservation.repository.ReservationRepository;
+import roomescape.domain.reservation.repository.ThemeRepository;
+
+@Service
+public class ThemeService {
+
+    private static final int START_DATE_OFFSET = 8;
+    private static final int END_DATE_OFFSET = 1;
+
+    private final Clock clock;
+    private final ThemeRepository themeRepository;
+    private final ReservationRepository reservationRepository;
+
+    public ThemeService(Clock clock, ThemeRepository themeRepository, ReservationRepository reservationRepository) {
+        this.clock = clock;
+        this.themeRepository = themeRepository;
+        this.reservationRepository = reservationRepository;
+    }
+
+    public List<ThemeResponse> getAll() {
+        return themeRepository.findAll()
+                .stream()
+                .map(ThemeResponse::from)
+                .toList();
+    }
+
+    public ThemeResponse create(ThemeRequest request) {
+        Theme theme = themeRepository.save(Theme.withoutId(request.name(), request.description(), request.thumbnail()));
```

Comment:
```suggestion
        Theme theme = themeRepository.save(request.toEntity());
```
생성에 대한 책임을 위임해보면 어떨까요?

Reply:
DTO에 엔티티 생성 책임을 전가하면 잘못된 생성 실수를 방지할 수 있다고 생각합니다

request를 entity로 변환하는 곳은 여러 곳에 퍼져 있어, 규모가 커질수록 관리해야 할 곳이 많아지기 때문에 dto에 엔티티 생성 책임을 두면 한 곳으로 책임을 모아 더 관리하기 수월해질 거라 생각합니다